src/lib.rs

```diff
@@ -104,62 +116,21 @@ impl FxHasher {
 impl Hasher for FxHasher {
     #[inline]
     fn write(&mut self, mut bytes: &[u8]) {
-        #[cfg(target_pointer_width = "32")]
-        let read_usize = |bytes: &[u8]| u32::from_ne_bytes(bytes[..4].try_into().unwrap());
-        #[cfg(target_pointer_width = "64")]
-        let read_usize = |bytes: &[u8]| u64::from_ne_bytes(bytes[..8].try_into().unwrap());
-
-        let mut hash = FxHasher { hash: self.hash };
-        assert!(size_of::<usize>() <= 8);
-        while bytes.len() >= size_of::<usize>() {
-            hash.add_to_hash(read_usize(bytes) as usize);
-            bytes = &bytes[size_of::<usize>()..];
+        const _: () = assert!(size_of::<usize>() <= size_of::<u64>());
```

Comment:
```suggestion
        const _: () = assert!(size_of::<u32>() <= size_of::<usize>() && size_of::<usize>() <= size_of::<u64>());
```

Or else the `u32::from_ne_bytes(u32_bytes) as usize` is not valid.

Reply:
Yup, good point